src/Token/Parser.php

```diff
@@ -90,7 +91,30 @@ private function parseHeader(string $data): array
      */
     private function parseClaims(string $data): array
     {
-        return (array) $this->decoder->jsonDecode($this->decoder->base64UrlDecode($data));
+        $claims = (array) $this->decoder->jsonDecode($this->decoder->base64UrlDecode($data));
+
+        if (isset($claims[RegisteredClaims::AUDIENCE])) {
```

Comment:
Note that casting `null` to `(array)` yields an empty array. I don't know if that's valid for `$claims`, but worth considering